python/ray/_private/node.py

```diff
@@ -206,8 +185,39 @@ def __init__(
                     f"{ray_params.dashboard_host}:{ray_params.dashboard_port}"
                 )
 
+        # It creates a session_dir.
         self._init_temp()
 
+        node_ip_address = ray_params.node_ip_address
+        if connect_only:
+            self.wait_for_node_address()
+        else:
+            if node_ip_address is None:
+                node_ip_address = ray._private.services.resolve_ip_for_localhost(
+                    ray_constants.NODE_DEFAULT_IP
+                )
+
+        node_ip_address = self._get_or_write_node_address(node_ip_address)
```

Comment:
I think it's cleaner to separate this into `get_node_address` and `write_node_address`, one is called for connect_only and the other is called by not_connect_only